src/workflow/factory.ts

```diff
@@ -33,6 +33,7 @@ export class WorkflowEngineFactory {
       optionsInteranl.currentBindings = bindings;
       const executor = await WorkflowUtils.getExecutor(workflow, optionsInteranl);
       const stepExecutors = await this.createStepExecutors(workflow.steps, optionsInteranl);
```

Comment:
Doesn't this validation get triggered throughout event processing flow whenever workflow is getting initialised ?
Instead shouldn't this go as a testcase in rudder-transformer ?

Reply:
this gets triggered only once during the initialization and it will be cached so it should be fine